qmt/geometry/part_3d.py

```diff
@@ -0,0 +1,285 @@
+
+    def write_stp(self, file_path=None):
+        """Write part geometry to a STEP file.
+
+        Returns the STEP file path.
+
+        Parameters
+        ----------
+        file_path : str
+            (Default value = None)
+        Returns
+        -------
+        file_path
+
+        """
+        if file_path is None:
+            file_path = f"{self.label}.stp"
+        write_deserialised(self.serial_stp, file_path)
+        return file_path
+
+    def write_stl(self, file_path=None):
+        """Write part geometry to a STEP file.
+
+        Returns the STEP file path.
+
+        Parameters
+        ----------
+        file_path : str
+            (Default value = None)
+        Returns
+        -------
+        file_path
+
+        """
+        if file_path is None:
+            file_path = f"{self.label}.stl"
+        write_deserialised(self.serial_stl, file_path)
+        return file_path
+
+
+class ExtrudePart(Geo3DPart):
+    def __init__(
+        self,
+        label: str,
+        fc_name: str,
+        thickness: float,
+        z0: float = 0.0,
+        virtual: bool = False,
+    ):
+        """Class for geometric extrusions.
+
+        Parameters
+        ----------
+        label : str
+            The descriptive name of this new part.
+        fc_name : str
+            The name of the 2D/3D freeCAD object that this is built from. Note that if the
+            label used for the 3D part is the same as the freeCAD label, and that label is
+            unique, None may be used here as a shortcut.
+        thickness : float
+            The extrusion thickness.
+        z0 : float
+            The starting z coordinate.
+            (Default value = 0.0)
+        virtual : bool
+            Whether the part is virtual or not.
+            (Default value = False)
+        """
+        self.thickness = thickness
+        self.z0 = z0
+        super().__init__(label, fc_name, virtual=virtual)
+
+
+class WirePart(Geo3DPart):
+    def __init__(
+        self,
+        label: str,
+        fc_name: str,
+        thickness: float,
+        z0: float = 0.0,
+        virtual: bool = False,
+    ):
+        """Class for hexagonal wire.
+
+        Parameters
+        ----------
+        label : str
+            The descriptive name of this new part.
+        fc_name : str
+            The name of the 2D/3D freeCAD object that this is built from. Note that if the
+            label used for the 3D part is the same as the freeCAD label, and that label is
+            unique, None may be used here as a shortcut.
+        thickness : float
+            The wire thickness.
+        z0 : float
+            The starting z coordinate.
+            (Default value = 0.0)
+        virtual : bool
+            Whether the part is virtual or not.
+            (Default value = False)
+        """
+        self.thickness = thickness
+        self.z0 = z0
+        super().__init__(label, fc_name, virtual=virtual)
+
+
+class DepoMode(Enum):
```

Comment:
Why is this its own class? It could even be just a bool.

Reply:
I thought it was good idea to use Enums (this is the exact purpose they're built for), but I guess their implementation in python is a bit clunky. I changed this to just a string